### DISCUSSION TITLE
Keeping environment variables out of repo

### DISCUSSION THREAD

**Discussion by ioionu** [piku/piku#302] [Q&A]
sorry if i missed this somewhere else.

just wondering what the best way to keep (some) environment variables out of the repo is?

i'm currently putting my top secret uber secrets in a separate `.env.prod` file and running:

```
piku config:set `cat .env.prod`
```

add `.env.prod` to `.gitignore`.

seems to work, but just thought id check and see if there is a better way.

**Comment by mariusa** [comment]
In `.env.prod`, do you keep all vars in one line? That command doesn't work with one var per line.

**Reply by ioionu** [reply]
using multi line env file, as far as i can tell its working, but entirely possible im missing something?:

```
cat env_file 
HELLO=WORLD
VAR2=BLA
VAR3=DERP

piku config:get VAR2
(snip)
BLA

> piku config:get VAR3
(snip)
DERP

```

**Comment by chr15m** [comment]
@ioionu that's a cool way to import multi-line secrets, thanks for sharing. Normally I have a crusty bash script for doing this but I'm going to start using your technique.

An interesting mid-way point is storing encrypted secrets in the repo: https://github.com/sobolevn/git-secret

**Comment by rcarmo** [comment]
See #304. I am looking into this, although TBH I favor storing secrets in external vaults (like Hashicorp’s or Azure Key Vault).
